### ISSUE TITLE
Fix S3242: Rule should not suggest base type resulting in inconsistent accessibility (bis)

### ISSUE THREAD

**Issue by antoinebj** [SonarSource/sonar-csharp#1610]
### Description

This is similar to #863, for which the fix apparently did not cover all situations.

### Repro steps

1. Create a solution with 2 projects

2. In one project, add the following code:
```csharp
internal interface IFoo
{
    bool IsFoo { get; }
}

public class Foo : IFoo
{
    public bool IsFoo { get; set; }
}
```

3. In the other project, reference the project that defines `Foo`, and add the following code:
```csharp
internal class Bar
{
    public void MethodOne(Foo foo)
    {
        var x = foo.IsFoo;
    }
}
```

### Expected behavior

A violation of S3242 should not be raised, because `IFoo` is not accessible from the project defining Bar.

### Actual behavior

A violation of S3242 is raised.
Note: If the `Bar` class is made public, that does not happen.

### Related information

* SonarC# Version 7.3
* Visual Studio Version 15.7
